flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java

```diff
@@ -64,15 +66,25 @@
 	private transient int commentCount;
 
 	private transient int invalidLineCount;
+
+	private PojoTypeInfo<OUT> pojoTypeInfo = null;
 	
 	
 	public CsvInputFormat(Path filePath) {
 		super(filePath);
-	}	
+	}
+
+	public CsvInputFormat(Path filePath, PojoTypeInfo<OUT> pojoTypeInfo) {
+		super(filePath);
+
+		Preconditions.checkNotNull(pojoTypeInfo, "The TypeInformation is required for getting the POJO fields.");
+		this.pojoTypeInfo = pojoTypeInfo;
+		setAccessibleToField();
+	}
 	
 	public CsvInputFormat(Path filePath, Class<?> ... types) {
```

Comment:
Yes, if we go for the more general TypeInformation we need to check whether it is a TupleTypeInfo or PojoTypeInfo and can get rid of one pair of constructors. Actually, I think both solutions are fine.